### ISSUE TITLE
Failed to run CNTK on Windows HPC 2012 R2 Using Multiple Compute Nodes

### ISSUE THREAD

**Issue by yefeng-zheng** [Microsoft/CNTK#166]
I am using the Windows version of CNTK. I can run it a computer with multiple GPUs. And, I can even run it on two computers not part of our Windows HPC. I start smpd or msmpilauchsvc on both computers and then run the following command on either computer. Everything works well.
mpiexec -hosts 2 MachineA 1 MachineB 1 CNTK.exe configFile=XXXX

However, I couldn’t run it on our Windows Compute Cluster. I got the following error message.
ERROR: Failed RpcCliStartMgr error -2147024809
Aborting: mpiexec on MachineA is unable to connect to the smpd service on MachineB:8677
Other MPI error, error stack:
connect failed - The parameter is incorrect.  (errno -2147024809)

I then tried to manually start smpd (running smpd -d). However, I got an access denied error. Do we really need to manually start smpd on each compute node of a cluster? 

Any suggestions to fix the issue? Thank you very much!

**Comment by Alexey-Kamenev** [comment]
Hello,
If you are running on Windows HPC then you should NOT use smpd. Win HPC already has its own service for MPI apps (msmpi) so you should run HPC MPI job rather than mpiexec directly.

**Comment by yefeng-zheng** [comment]
Yes, this was also my understanding. I think Windows HPC uses msmpi service instead of smpd. 

However, when I submit the job. I got an error as "mpiexec on MachineA is unable to connect to the smpd service on MachineB:8677." CNTK tries to use smpd for the communication between nodes, instead of using msmpi. Do you have how to fix?

**Comment by Alexey-Kamenev** [comment]
Are you sure your HPC cluster is setup correctly? For example, firewall exceptions etc?

**Comment by yefeng-zheng** [comment]
The HPC cluster has been running for a while. There is no problem at all. The MSMPI-launchsvc, MSMPI-mpiexec, and MSMPI-smpd are already in the firewall exceptions. I also added CNTK to the exceptions. But, it didn't fix the problem.

I tried to turn off the firewall completely, following https://technet.microsoft.com/en-us/library/dd772588%28v=ws.10%29.aspx?f=255&MSPPError=-2147217396. But, after a few minutes, the firewall was turned on automatically.
